x-pack/plugins/timelines/public/components/actions/timeline/cases/add_to_new_case_button.tsx

```diff
@@ -25,22 +27,36 @@ const AddToNewCaseButtonComponent: React.FC<AddToNewCaseButtonProps> = ({
   owner,
   onClose,
 }) => {
-  const { addNewCaseClick, isDisabled, userCanCrud } = useAddToCase({
+  const { isDisabled, userCanCrud, caseAttachments, onCaseSuccess, onCaseCreated } = useAddToCase({
     event,
     useInsertTimeline,
     casePermissions,
     appId,
     owner,
     onClose,
   });
+  const { cases } = useKibana<TimelinesStartServices>().services;
+  const createCaseFlyout = cases.hooks.getUseCasesAddToNewCaseFlyout({
+    attachments: caseAttachments,
+    afterCaseCreated: onCaseCreated,
+    onSuccess: onCaseSuccess,
+  });
+
+  const handleClick = () => {
```

Comment:
If I have understood correctly in future PRs cases will provide the `EuiContextMenuItem` or similar and the logic of closing/opening the flyout will be handled by cases. Is that correct?

Reply:
Exactly. this code will move out of here in about two PRs.